File: java/src/main/java/mx/iteso/adapter/impl/ints/IGoogleDoc.java
package mx.iteso.adapter.impl.ints;

import java.awt.Font;
import mx.iteso.adapter.impl.other.BackgroundImage;
/**
 * Google document interface.
 */
public interface IGoogleDoc {
    /**
     * Gets Font.
     * @return font.
     */
    Font getFont();
    /**
     * @return object of style.
     */
    Object getStyle();
    /**
     * @return Background Image of document.
     */
    BackgroundImage getBackground();
    /**
     *
     * @param sharingPermissions sets permissions.
     */
    void setSharingPermissions(int sharingPermissions);
}

File: csharp/AdapterPractice/AdapterPractice/Adapter/GoogleDocs/IGoogleDoc.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AdapterPractice.Adapter.GoogleDocs
{
    public interface IGoogleDoc
    {

        Font getFont();


        Format getStyle();


        BackgroundImage getBackground();


        void setSharingPermissions(int sharingPermissions);


        int getSharingPermissions();
    }
}
